src/function/scalar/string/levenshtein.cpp

```diff
@@ -0,0 +1,84 @@
+#include "duckdb/function/scalar/string_functions.hpp"
+#include "duckdb/common/vector_operations/vector_operations.hpp"
+#include "duckdb/common/string_util.hpp"
+
+#include <ctype.h>
+#include <algorithm>
+
+namespace duckdb {
+
+// See: https://www.kdnuggets.com/2020/10/optimizing-levenshtein-distance-measuring-text-similarity.html
+// And: Iterative 2-row algorithm: https://en.wikipedia.org/wiki/Levenshtein_distance
+// Note: A first implementation using the array algorithm version resulted in an error raised by duckdb (too muach memory usage) 
+static int levenshtein2(const string& txt, const string& tgt) 
```

Comment:
We prefer to avoid the use of `std::string` since it performs a large number of (unnecessary) allocations. Could you change this function to operate on our internal `string_t` type instead? The resulting code would look very similar, only using `GetSize()` instead of `size()`, and `GetDataUnsafe()` to obtain a raw pointer to the data (note that string_t is not null-terminated).

Reply:
Thanks for the pointers.